server/controllers/stock/reports/stock/value.js

```diff
@@ -30,21 +30,109 @@ async function reporting(_options, session) {
 
   const report = new ReportManager(STOCK_VALUE_REPORT_TEMPLATE, session, optionReport);
 
+
   const options = (typeof (_options.params) === 'string') ? JSON.parse(_options.params) : _options.params;
   data.dateTo = options.dateTo;
   data.depot = await db.one('SELECT * FROM depot WHERE uuid=?', [db.bid(options.depot_uuid)]);
-  const stockValues = await db.exec('CALL stockValue(?,?,?);', [
-    db.bid(options.depot_uuid), options.dateTo, options.currency_id,
-  ]);
+
+  // Get inventories movemented
+  const sqlGetInventories = `
+    SELECT DISTINCT(BUID(mov.inventory_uuid)) AS inventory_uuid, mov.text AS inventory_name
+    FROM(
+      SELECT inv.uuid AS inventory_uuid, inv.text, sm.date
+      FROM stock_movement AS sm
+      JOIN lot AS l ON l.uuid = sm.lot_uuid
+      JOIN inventory AS inv ON inv.uuid = l.inventory_uuid
+      WHERE sm.depot_uuid = ? AND DATE(sm.date) <= DATE(?)
+    ) AS mov
+    ORDER BY mov.text ASC;
+  `;
+
+  /*
+   * The following query collects all the data related to the movement
+   * of stocks of an item in order to calculate the stock value
+   * With the aim of solving the problem, count the total number of
+   * entries into stock *quantityIn* as well as that of exit
+   * but also the cumulative value of entries *quantityCum* and exits *cumOut*
+   * and the quantity in stock after each movement of stock *quantityStock*
+  */
+  const stockValues = await db.exec(sqlGetInventories, [db.bid(options.depot_uuid), options.dateTo]);
```

Comment:
Sp, if I am understanding this correctly, this code will generate `1 + N` SQL queries in a transaction, where `N` is the number of inventory items in a depot.  At Vanga, the total number of inventory items is currently 784.  At IMCK, the total number is 1302.  So we could potentially get hundreds of SQL queries launched from a single query here.

... I'm not sure how optimal this is in low-end hardware like a Raspberry Pi.

Reply:
the way I understood the logic of the stock value report came down to retrieving the last line of the stock card report, my idea was to launch several requests on the stock movement table in order to calculate for each unit cost weight and the stock value after each transaction

Reply:
I had given up on the idea of using an aggregation table because of the risk of having backdated stock movements